automation/module_eval/src/evaluators/java/shared-evaluator.ts

```diff
@@ -0,0 +1,179 @@
+
+  private async evaluateS001(repoPath: string): Promise<CriterionResult> {
+    return await LicenseUtils.checkApache2License(repoPath, 'S001');
+  }
+
+  private async evaluateS002(_repoPath: string): Promise<CriterionResult> {
+    return this.createManualReviewResult('S002', 'Module descriptor validation - stub implementation');
+  }
+
+  private async evaluateS003(repoPath: string): Promise<CriterionResult> {
+    try {
+      // Extract dependencies from the repository
+      const dependencies = await getDependencies(repoPath);
+      
+      if (dependencies.length === 0) {
+        return {
+          criterionId: 'S003',
+          status: EvaluationStatus.MANUAL,
+          evidence: 'No dependencies found or unable to extract dependency information',
+          details: 'Could not analyze third-party dependencies. This may be due to build tool configuration issues or the repository not containing dependency information.'
+        };
+      }
+
+      // Read README content for Category B license validation
+      let readmeContent = '';
+      const readmeFiles = ['README.md', 'README.txt', 'README', 'readme.md', 'readme.txt'];
+      
+      for (const readmeFile of readmeFiles) {
+        const readmePath = path.join(repoPath, readmeFile);
+        if (fs.existsSync(readmePath)) {
+          readmeContent = fs.readFileSync(readmePath, 'utf-8');
+          break;
+        }
+      }
+
+      // Check license compliance according to ASF policy
+      const complianceResult = checkLicenseCompliance(dependencies, readmeContent);
+      
+      // Generate evidence summary
+      const dependencyCount = dependencies.length;
+      const licenseInfo = dependencies
+        .filter(d => d.licenses && d.licenses.length > 0)
+        .map(d => `${d.name}:${d.version} (${d.licenses!.join(', ')})`)
+        .join('; ');
+      
+      const evidence = `Found ${dependencyCount} dependencies. ` +
+        (licenseInfo ? `Licenses: ${licenseInfo}` : 'No license information available for analysis.');
+
+      if (complianceResult.compliant) {
+        return {
+          criterionId: 'S003',
+          status: EvaluationStatus.PASS,
+          evidence: evidence,
+          details: 'All third-party dependencies comply with ASF 3rd Party License Policy. No Category X licenses found, and all Category B licenses are properly documented.'
+        };
+      } else {
+        // Generate detailed compliance issues
+        const issueDetails = complianceResult.issues
+          .map(issue => `• ${issue.dependency.name}:${issue.dependency.version} - ${issue.reason}`)
+          .join('\n');
+
+        return {
+          criterionId: 'S003',
+          status: EvaluationStatus.FAIL,
+          evidence: evidence,
+          details: `Third-party license compliance issues found:\n${issueDetails}\n\nPlease resolve these issues according to ASF 3rd Party License Policy.`
```

Comment:
`issueDetails` will be multiple lines long in some cases.  Better to put all of the text before the variable, i.e. 

> Third-party license compliance issues found.  Please resolve these issues according to ASF 3rd Party License Policy.  :\n${issueDetails}\n\n

<img width="1184" height="76" alt="image" src="https://github.com/user-attachments/assets/4047cb36-a374-4c0b-9078-040d1e751fe7" />